src/OutputWriters/jld2_output_writer.jl

```diff
@@ -158,33 +158,36 @@ JLD2OutputWriter scheduled on TimeInterval(20 minutes):
 └── max filesize: Inf YiB
 ```
 """
-function JLD2OutputWriter(model, outputs; prefix, schedule,
+function JLD2OutputWriter(model, outputs; filename, schedule,
                                    dir = ".",
                                indices = (:, :, :),
                             with_halos = false,
                             array_type = Array{Float32},
                           max_filesize = Inf,
-                                 force = false,
+                    overwrite_existing = false,
                                   init = noinit,
                              including = default_included_properties(model),
                                verbose = false,
                                   part = 1,
                                jld2_kw = Dict{Symbol, Any}())
 
+    # Enforce that extensions need to be `jld2`
+    @assert filename[end-4:end] == ".jld2" "`filename` needs to have `.jld2` extension"
```

Comment:
```suggestion
    filename[end-4:end] == ".jld2" || throw(ArgumentError("`filename` needs to have `.jld2` extension"))
```

Reply:
What's wrong with assert? Do you want an `ArgumentError`?

Reply:
Just a little more friendly... `AssertionError` is sometimes mysterious (the purpose of `@assert` is to provide a correctness check that might be undesirable for performance, so gets removed at a certain optimization level)

Reply:
Here's the docstring for `@assert`:

```julia
help?> @assert
  @assert cond [text]

  Throw an AssertionError if cond is false. Preferred syntax for writing assertions. Message text is optionally displayed upon assertion failure.

  │ Warning
  │
  │  An assert might be disabled at various optimization levels. Assert should therefore only be used as a debugging tool and not used for authentication verification (e.g., verifying passwords), nor
  │  should side effects needed for the function to work correctly be used inside of asserts.
```

Reply:
`ArgumentError` means

```julia
help?> ArgumentError
search: ArgumentError

  ArgumentError(msg)

  The parameters to a function call do not match a valid signature. Argument msg is a descriptive error string.
```

Here we are telling the user that the argument `filename` is incorrect.

Reply:
You might envision here that somebody attempts to run a model at `O3` optimization, with a bug in `filename`. Then hours into the simulation they get an error (maybe) because we assume `.jld2`.